generators/php/codegen/src/context/PhpAttributeMapper.ts

```diff
@@ -41,14 +45,41 @@ export class PhpAttributeMapper {
             attributes.push(
                 php.attribute({
                     reference: this.context.getArrayTypeClassReference(),
-                    arguments: [this.getArrayTypeAttributeArgument(type.underlyingType())]
+                    arguments: [this.getTypeAttributeArgument(type.underlyingType())]
                 })
             );
         }
+        if (underlyingInternalType.type === "union") {
+            const unionTypeParameters = this.getUnionTypeParameters(underlyingInternalType.types);
+            // only add the attribute if deduping in getUnionTypeParameters resulted in more than one type
+            if (unionTypeParameters.length > 1) {
+                attributes.push(
+                    php.attribute({
+                        reference: this.context.getUnionClassReference(),
+                        arguments: this.getUnionTypeParameters(underlyingInternalType.types)
+                    })
+                );
+            }
+        }
         return attributes;
     }
 
-    public getArrayTypeAttributeArgument(type: php.Type): php.AstNode {
+    public getUnionTypeClassRepresentation(arguments_: php.AstNode[]): ClassInstantiation {
```

Comment:
nit: 
```suggestion
    public getUnionTypeClassInstantiation(arguments_: php.AstNode[]): ClassInstantiation {
```

Ideally the argument could be a little more strongly typed (e.g. `php.Type`). Otherwise it's possible to write in arbitrary `php.CodeBlock` here. But I know it's difficult because of the `getTypeAttributeArgument` helper function below - the attributes aren't the same as the type representation (e.g. `int` vs `integer`).

Fine with this for now, but we'll need to be careful to call this with arguments that make sense.